### ISSUE TITLE
Bug: messageformat not transpiled when selecting an object with selectTranslate

### ISSUE THREAD

**Issue by fxck** [ngneat/transloco#47]
See console of 

https://stackblitz.com/edit/angular-aa3pmd?file=src/app/app.component.ts

@NetanelBasal 

...btw I have a feeling that `Translation[]` is a valid return type as well `selectTranslate<T = string | Translation | string[]>`

**Comment by fxck** [comment]
Ok, it wasn't exactly working this way in ngx-translate either, if you used messageformat compiler, it returned a function under the `bar` property, which you would pass the params to 

https://stackblitz.com/edit/github-idytfc-ayovbm?file=src/app/app.component.ts

can we perhaps do this better somehow?

**Comment by fxck** [comment]
Played around with it a bit, see 

https://github.com/ngneat/transloco/compare/master...fxck:ar-nestedmessageformattranspilation

what would be required

- transpile to return `string | Translation` (which I think is correct either way, since [currently](https://github.com/ngneat/transloco/commit/ff835c38af5c0e59647245039d1e951c0548f244#diff-e4be93846cf644f2366c0e93911d139cR14) it indeed might return something else other than string)
- in case an object was passed to messageformat transpiler, it needs to go thru it and compile everything
  - what I've done is a recursive iteration, which could be costly, it also requires all the params to be unique
  - what I think could be a better solution would be to pass sort of a nested map of params, ie.

```typescript
      this._translocoService
        .selectTranslate('mf.nested', {
          people: { count: '1' },
          projects: { moreNesting: { count: '1' } }
        })
```

               that way we would know which nested properties need to be compiled

**Comment by fxck** [comment]
Here's proof of concept of the alternative solution with string path and `set`

https://github.com/ngneat/transloco/compare/master...fxck:ar-nestedmessageformattranspilation2

**Comment by NetanelBasal** [comment]
We are moving `messageformat` to a separate package. Let's discuss this issue afterward.

**Comment by NetanelBasal** [comment]
@shaharkazaz will make the first suggestion both to our transpiler and the `messgaeforamt`.

**Comment by fxck** [comment]
Any thoughts so far @shaharkazaz? I was thinking and maybe what ngx-translate is doing might be useful in some cases as well, maybe it could be optional along params object?

**Comment by NetanelBasal** [comment]
> useful in some cases as well

For example?

**Comment by fxck** [comment]
I'm passing translation down to a component, which displays number of people and projects, so I'm passing those numbers down to the anyway (to display the number, which is not part of the person/people, project/projects translation), so instead of having to combine those number and passing them to selectTranslations, I'd just do `{{ traslations.projects(projectCount) }}` inside the template of the component.

**Comment by fxck** [comment]
But I think I still like having it resolved beforehand better.

**Comment by NetanelBasal** [comment]
We should choose one way for now.

**Comment by NetanelBasal** [comment]
LNK, If you want to take it and create a PR that will support both.

**Comment by NetanelBasal** [comment]
I just noticed that it returns a function for any value, even if it is a simple string. https://stackblitz.com/edit/github-idytfc-7cgt4g

Not so intuitive.

**Comment by fxck** [comment]
I think their translate directive unwraps it or something. Anyway personally I'd go for this https://github.com/ngneat/transloco/compare/master...fxck:ar-nestedmessageformattranspilation2 only, for now anyway.

**Comment by NetanelBasal** [comment]
BTW, from what I see, ngx-translate (not related to `messageformat`) doesn't support object translation with params: https://stackblitz.com/edit/github-idytfc-7cgt4g

**Comment by NetanelBasal** [comment]
Fixed in 1.7.0
